SWIPs/swip-honey_to_money.md

```diff
@@ -9,29 +9,29 @@ Created: 31-07-2019
 ---
 
 ## Simple summary 
-Swarm needs a mechanism to allow fluctuation of the exchange rate between honey (Swarms internal accounting unit) and the currency used to pay. We propose an on-chain price oracle that returns an exchange rate + validUntil tuple.  
+Swarm needs a mechanism to allow fluctuation of the exchange rate between honey (Swarms internal accounting unit) and the currency used to pay. We propose an on-chain price oracle that returns an `(exchange rate,validUntil)` tuple.  
 
 ## Abstract 
 The ability to update prices is required in order to:
 * Facilitate experimentation with the absolute price of honey (e.g. start at a low price and gradually increase);
-* Off-set fluctuations of the currency used to pay;
-* Lower the price of Swarm in the long-run as the costs of operating a Swarm node decreases.
+* Offset fluctuations of the currency used to pay;
```

Comment:
I am not sure I agree to this. I don't think  we should actually do this. We should try to avoid importing external currency fluctuation discourse into this by all means IMHO. We should not actively try to counter-balance currency market fluctuations. What is the intention to have this?

Also, think of the complexity if we support multiple currency; would we have to offset fluctuations of every currency? I hope we do not.

Reply:
It would very undesirable that, when Ether doubles in price overnight, the price of using Swarm also doubles overnight without the possibility to re-adjust. Hence, I think it is absolutely to be able to adjust prices based on currency fluctuations. However, oracle-models can be thought of where price fluctuations are priced-in without manual intervention, such as letting the honey-money oracle read from the [MAKER ETH/USD Oracle](https://github.com/makerdao/developerguides/blob/master/mcd/mcd-101/mcd-101.md). And let the final honey price be defined by a function such as:

`HoneyPrice = BaseFee + PriceSensitivity*EthExcangeRate + otherSensitivities*otherFactors` 

More research is definitely needed here, but the oracle model facilitates upgrading, which means we don't have to have it right the first time. And for the first iteration, setting everything manually will probably be good enough.

In the case with multiple currencies: as I see it now, Swarm will only maintain the honey/wei oracle and other oracles will be maintained by the groups who choose to add this currency to Swarm as a payment module, they will be incentivized to follow the default oracle (the Swarm Honey/wei) oracle: when they quote an exchange rate which is too high, only net contributors of Swarm will prefer their oracle, when they quote a price which is too low, only net user of Swarm will prefer their oracle => this oracle will not be preferred by any two nodes where one is a net user and the other is a net contributor.

Reply:
We should want to adapt prices so that the costs of using swarm and the costs of running swam reflect actual real-world costs. Eknir is right to point out that using ether instead of using honey and allowing honey-to-ether to fluctuate, exposes swarm to severe price swings that could harm the network economically.

We don't want to offset fluctuations for every currency.

We translate Honey To Ether (any try to keep the rates somewhat constant (perhaps tied to the DAI price oracle).. but all further currencies and oracles are not our concern. Anyone can run any oracle and any two Swarm nodes can agree to allow any one of them in their settlements. The nice thing about oracles is that they are external.

Note: While gas costs are hard coded, but gas price fluctuates. We need similar flexibility so we cannot just hard code ether prices - that would be like hardcoding transaction costs in ether.

Reply:
So is someone to watch coinmarketcap daily and adjusting the price? How is a price being deployed? Is it a single person watching the swings and then unilaterally deciding on adjusting the price? If there is some governance behind it, then it will be too slow?

If there is no governance, then it will be perceived as abuse of power. Who has the fine-tuning knowledge to adjust prices correctly? Is it a research meeting every week? Too slow to counteract the real currency fluctuations. Every day? Is it one person? Too risky that one person has to decide.

In any case there is the risk of actually harming even more than what we think we are fixing by changing prices to counteract market fluctuations; there is the risk to over/undershoot persistently. Crypto is highly volatile and that's the nature of it, like it or not. And it is in my personal opinion the biggest threat and misunderstanding in crypto, in that it doesn't set the value but is defined in dependency to fiat, which makes it volatile and subject to speculation.

In my opinion, if the currency is ETH, then the fluctuations are on ETH, not on Swarm. If you price your hosting in EUR, and the EUR fluctuates (hypothetically) against USD, do you adjust your hosting in EUR? No you don't because that is your local currency.

The risk is on ETH, not on Swarm, if ETH gains/looses values it's ETH not Swarm. 

I am not convinced by these arguments and still think this should not be included. Of course we have the possibility to adjust prices, that's what the oracle is for, but I don't think it should be to counteract market fluctuations, but to set prices which makes sense *relative to other services in the ETH market*

Reply:
@holisticode, if you say that you want the price of Swarm to be sensible relative to other services in the ETH market, I think we agree that Swarm prices *must* be adjusted to offset currency fluctuations; any service which is priced will -everything else equal- halve in price (denominated in ETH), when the ETH price doubles in price (denominated in one of the major other currencies). 

Your point on governance is a good one, but you have to realize that even saying that there will be *no* offset of currency fluctuations is a decision and hence does not free you from having a proper form of governance. What would free us from governance is that prices will be set entirely peer-to-peer, but since Swarm requires peers to connect with each other and because there is no way to implement a price-finding mechanism yet, we have proposed the oracle. 

I would propose to write something like this if it makes you feel more comfortable:
```
The ability to update prices is required as Swarm needs to be priced sensibly relative to other services in the (ETH) market and the costs to operate a node. Furthermore, the price must be adjustable to facilitate experimentation.
```